### PR TITLE
add  ...  as param in query favorite

### PR DESCRIPTION
**Pull request by fraoustin**
## Description

same working between mycli and pgcli about substiting parameters in favorite queries.

sample of usage

```
> create table test(id integer, lib varchar(25)
> insert into test(id,lib) values (1,'test')
> \ns test select * from test where id=$1
> \n test 1
+------+-------+
| id   | lib   |
|------+-------|
| 1    | test  |
+------+-------+
> \ns test select * from test where id=$1
> \n test 1
+------+-------+
| id   | lib   |
|------+-------|
| 1    | test  |
+------+-------+
```

## Checklist

- [ ] I've added this contribution to the `changelog.rst`.

### CODE REVIEW

**Comment by j-bennet** [comment]
@fraoustin Thank you for the contribution! Your unit tests fail on pep8 checker. Run `pep8radius master --docformatter -i` to fix the formatting problem.

**Comment by fraoustin** [comment]
it should be fine

**Comment by j-bennet** [comment]
@fraoustin Looks good to me! Please add your contribution to the changelog, and I'll merge it.

**Comment by fraoustin** [comment]
It's done thank you

File: pgspecial/iocommands.py [hunk 1]
```diff
@@ -168,6 +183,8 @@ def execute_named_query(cur, pattern, **_):
         return [(None, None, None, message)]
 
     try:
+        if "$1" in query:
+            query, params = subst_favorite_query_args(query, params)
```
Comment (j-bennet) [inline pgspecial/iocommands.py]: Found a bug here. In case of missing substitution, you return `None` for the query. So you need a check for `if query is None`, and raise an error if that happens. Right now you'll get `can't execute an empty query` error, and that's not correct.

**Comment by fraoustin** [comment]
add test on "query is None" raise Exception as bad arguments

File: pgspecial/iocommands.py (hunk 1 above)
Comment (fraoustin) [reply pgspecial/iocommands.py]: add test on "query is None" raise Exception as bad arguments

**Comment by fraoustin** [comment]
It's ok for you?

**Comment by j-bennet** [comment]
All good now.

Thank you for the PR!

🍫
